src/panels/lovelace/entity-rows/hui-climate-entity-row.js

```diff
@@ -0,0 +1,107 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../components/hui-generic-entity-row.js';
+
+import LocalizeMixin from '../../../mixins/localize-mixin.js';
+
+/*
+ * @appliesMixin LocalizeMixin
+ */
+class HuiClimateEntityRow extends LocalizeMixin(PolymerElement) {
+  static get template() {
+    return html`
+      <style>
+        .state-label {
+          font-weight: bold;
+          text-transform: capitalize;
+        }
+      </style>
+      <hui-generic-entity-row
+        hass="[[hass]]"
+        config="[[_config]]"
+      >
+        <div>
+          <span class="state-label">
+            [[_computeState(_stateObj)]]
+          </span>
+          [[_computeTarget(_stateObj)]]
+        </div>
+        
+        <div slot="secondary">
+          [[localize('ui.card.climate.currently')]]: [[_computeCurrentStatus(_stateObj)]]
+        </div>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: Object,
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)'
+      }
+    };
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error('Entity not configured.');
+    }
+    this._config = config;
+  }
+
+  _computeState(stateObj) {
+    if (!stateObj) {
+      return this.localize('state.default.unavailable');
+    }
+    return this.localize(`state.climate.${stateObj.state}`) || stateObj.state;
+  }
+
+  _computeCurrentStatus(stateObj) {
+    if (!stateObj) return null;
+
+    if (stateObj.attributes.current_temperature != null) {
+      return `${stateObj.attributes.current_temperature} ${stateObj.attributes.unit_of_measurement}`;
```

Comment:
I think we should use `hass.config....` instead and deprecate this attribute for climate as it's for an attribute and not the state. This will also fix the localizeState function for climate

cc: @balloob

Reply:
Yes, I agree.